test/verify/check-metrics

```diff
@@ -839,65 +839,89 @@ podman save quay.io/libpod/busybox | sudo -i -u admin podman load
         b.wait_not_present("#current-metrics-card-cpu .temperature")
 
         # No matching type
-        m.execute("mkdir -p /tmp/hwmon/hwmon0")
-        m.execute("echo BAT0 > /tmp/hwmon/hwmon0/name")
-        m.execute("mount -o bind /tmp /sys/class")
+        m.execute("mkdir -p /tmp/sensor-sys-class/hwmon/hwmon0")
+        self.addCleanup(m.execute, "rm -rf /tmp/sensor-sys-class/hwmon")
+        m.write("/tmp/sensor-sys-class/hwmon/hwmon0/name", "BAT0")
+        m.write("/tmp/sensor-sys-class/hwmon/hwmon0/temp1_input", "40000")
+        m.execute("mount -o bind /tmp/sensor-sys-class /sys/class")
+        self.addCleanup(m.execute, "umount /sys/class")
 
         b.wait_not_present("#current-metrics-card-cpu .temperature")
 
-        # create files that contain CPU temperature
-        m.execute("mkdir -p /tmp/hwmon/hwmon1")
-
-        # Testing on ARM devices
-        m.execute("echo cpu_thermal > /tmp/hwmon/hwmon1/name")
-        m.execute("echo 60000 > /tmp/hwmon/hwmon1/temp1_input")
-
-        b.wait_in_text("#current-metrics-card-cpu", "60 °C")
+        def clean_and_relog():
+            m.execute("rm -rf /sys/class/hwmon/hwmon1/*")
+            b.relogin()
+            b.go("/metrics")
+            b.enter_page("/metrics")
```

Comment:
This looks like a race condition to me -- you already open the metrics page *without any sensors set up*. As the bridge only detects them once, ever, you currently hope that the page opens the channel slower than the test will create the fake files. Do your tests still work if you add a `time.sleep(3)` after this line here? If they do, then I don't understand how this can work at all.

You should better split the cleanup/logout and the relogin, and put the new /sys setup in  between for each scenario.

Reply:
[this failure](https://cockpit-logs.us-east-1.linodeobjects.com/pull-17601-20220729-050759-728e75ad-centos-8-stream/log.html#305-1) could very well be due to this race condition?